packages/bp/src/core/messaging/messaging-service.ts

```diff
@@ -40,15 +47,24 @@ export class MessagingService {
     const config = await this.configProvider.getBotConfig(botId)
     let messaging = (config.messaging || {}) as MessagingConfig
 
+    const webhookUrl = `${process.EXTERNAL_URL}/api/v1/chat/receive`
     const setupConfig = {
       name: botId,
-      ...messaging
-      // We use the SPINNED_URL env var for now to force the messaging server to
-      // make its webhook requests to the process that started it.
-      // webhooks: [{ url: `${process.EXTERNAL_URL}/api/v1/chat/receive` }]
+      ...messaging,
+      // We use the SPINNED_URL env var to force the messaging server to make its webhook
+      // requests to the process that started it when using a local Messaging server
+      webhooks: this.isExternal ? [{ url: webhookUrl }] : []
     }
 
-    const { id, token } = await this.clientSync.syncClient(setupConfig)
+    const { id, token, webhooks } = await this.clientSync.syncClient(setupConfig)
+
+    if (webhooks?.length) {
+      for (const webhook of webhooks) {
+        if (webhook.url === webhookUrl) {
+          this.webhookToken = webhook.token
+        }
+      }
+    }
```

Comment:
You should just do webhooks[0] here. If we didn't get back a webhook then there's an error somewhere anyway